mongo/integration/unified/testrunner_operation.go

```diff
@@ -262,6 +275,48 @@ func executeLoop(ctx context.Context, args *loopArgs, loopDone <-chan struct{})
 	}
 }
 
+type waitForEventArguments struct {
+	ClientID string              `bson:"client"`
+	Event    map[string]struct{} `bson:"event"`
+	Count    int32               `bson:"count"`
+}
+
+// eventCompleted will check all of the events in the event map and return true if all of the events have at least the
+// specified number of occurrences. If the event map is empty, it will return true.
+func (args waitForEventArguments) eventCompleted(client *clientEntity) bool {
+	for rawEventType := range args.Event {
+		eventType, ok := monitoringEventTypeFromString(rawEventType)
+		if !ok {
+			return false
+		}
+
+		if client.getEventCount(eventType) < args.Count {
+			return false
+		}
+	}
+
+	return true
+}
+
+func waitForEvent(ctx context.Context, args waitForEventArguments) error {
+	client, err := entities(ctx).client(args.ClientID)
+	if err != nil {
+		return err
+	}
+
+	for {
+		select {
+		case <-ctx.Done():
+			return fmt.Errorf("timed out waiting for event: %v", ctx.Err())
+		default:
```

Comment:
@benjirewis Unfortunately, we don't have access to a testing object in `unified.executeTestRunnerOperation`. 

@matthewdale What are your thoughts on following a similar pattern to the integration package analogue, [here](https://github.com/mongodb/mongo-go-driver/blob/master/mongo/integration/unified_runner_events_helper_test.go#L111)?

Reply:
The context provides this functionality:

```
time.Since(start) < waitForEventTimeout
```

I could definitely be wrong, but I think these are two different ways of writing the same logic. But, having the context also lends us access to the parents context, if for some reason this function needs that at some point in the future.

Reply:
The current solution looks good to me with the `time.Sleep`.

Reply:
Seems reasonable as long as it's not a tight loop.